docs/proposals/33-machine-image-capabilities.md

```diff
@@ -0,0 +1,639 @@
+This will display the version as supported in the Gardener Dashboard.
+This is not the intended use of the classification key.
+
+3. Currently extension providers define their own capabilities that are processed after shoot admission, like `acceleratedNetworking` in the [gardener-extension-provider-azure](https://github.com/gardener/gardener-extension-provider-azure/blob/28c977612898ed40e9d179052633fee0b9600d3e/pkg/apis/azure/types_cloudprofile.go#L78).
+This can lead to a situation where the machine supports accelerated networking but the machine image does not.
+In this case it does not result in an error but in performance loss.
+
+### Goals
+
+- Define a pattern to ensure only shoots with compatible machine types and images are admitted by Gardener.
+- Provide a mechanism to filter out incompatible images for a machine type in the Gardener Dashboard.
+- Provide a pattern and process to add new or remove deprecated capabilities in the future.
+- Minimize the additional data size introduced to cloud profiles to prevent bloating.
+- Obsolete existing workarounds described in the "Current Workarounds" section.
+
+### Non-Goals
+
+- Introduce capability options that are not utilized by Gardener in the future.
+- Introduce feature flag to control the usage of a capability, e.g. secure boot.
+
+## Proposal
+
+Introduce a top level capabilities map in the CloudProfile `spec.capabilities`.
+
+Capabilities are very specific to the provider and the selected catalog offered by Gardener in the cloud profile.
+To minimize complexity and data size in the cloud profile, the capabilities are defined as a map with string keys and string arrays as values.
+The key is the capability name and the value is an array of possible values.
+
+```go
+type Spec struct {
+    Capabilities map[string][]string `json:"capabilities"`
+}
+```
+
+For each cloud profile the capabilities are defined in the `spec.capabilities` map.
+The full set of possibilities for each capability is defined here.
+As some capabilities can have multiple values at the same time an array of possible values is used instead of a single value.
+
+This structure defines the default set of capabilities.
+If no further information are provided each machine type and machine image will be assigned the default set of capabilities.
+Thus being compatible with each other.
+
+```yaml
+# CloudProfile Example
+spec:
+  capabilities:
+    hypervisorType: ["gen2", "gen1"]
+    network: ["accelerated", "standard"]
+    storageAccess: ["NVMe", "SCSI"]
+    secureBoot: ["secure", "none"]
+    bootMode: ["uefi-preferred", "uefi", "legacy-bios"]
+    ...
+```
+
+The array of a capability is ordered.
+The order defines the priority of the values in case of multiple supported images.
+The first value in the array is the most preferred value.
+The last value is the least preferred value.
```

Comment:
```suggestion
Please note the following characteristics:
- The array of a capability is ordered.
- The order defines the priority of the values in case of multiple supported images.
- The first value in the array is the most preferred value.
- The last value is the least preferred value.
```